Readers/src/GenJetReader.cc

```diff
@@ -90,7 +90,6 @@ GenJetReader::setBranchAddresses(TTree * tree)
     bai.setBranchAddress(jet_eta_, branchName_eta_);
     bai.setBranchAddress(jet_phi_, branchName_phi_);
     bai.setBranchAddress(jet_mass_, branchName_mass_);
-    bai.setBranchAddress(jet_pdgId_, branchName_pdgId_);
```

Comment:
This branch is not available in nanoaod file /hdfs/local/snandan/hhAnalysis/2017/official_nanoaod_file/F1B4499C-4A59-D142-8C2E-8364C1C6F90Cii.root but  present in previous file /hdfs/local/karl/ttHNtupleProduction/2017/2021Mar05_wPresel_nom_all_hh_multilepton/ntuples/signal_ggf_nonresonant_node_sm_hh_4v/0000/tree_1.root. I see this branch is available https://github.com/HEP-KBFI/TallinnNtupleProducer/blob/main/Readers/src/GenJetReader.cc#L79 . May be we need to turn on read_partonFlavour_, if this is the case please let me know and I'll do that

Reply:
In the old software, `produceNtuple` used `GenJetReader` to read the branches but `GenParticleWriter` to write the collection. So, in order to read the collection back with `GenParticleReader` at the analysis level, the branch name had to be transcribed.

Since we don't have such problem anymore, I recommend that you read both partonFlavor and hadronFlavor of the gen jet, drop pdgId, and name the member variables accordingly.

Reply:
done here https://github.com/HEP-KBFI/TallinnNtupleProducer/pull/3/commits/acab57b16864ce89c73f994eee3e70e6915ef846